app/scripts/lib/tx-verification/tx-verification-middleware.ts

```diff
@@ -49,55 +60,55 @@ export function createTxVerificationMiddleware(
 
     // the tx object is the first element
     const params = req.params[0];
-
     const chainId =
       typeof params.chainId === 'string'
-        ? (params.chainId.toLowerCase() as `0x${string}`)
+        ? (params.chainId.toLowerCase() as Hex)
         : networkController.state.providerConfig.chainId;
 
-    // skip verification if bridge is not deployed on the specified chain.
-    // skip verification to address is not the bridge contract
-    if (
-      !Object.keys(FIRST_PARTY_CONTRACT_NAMES['MetaMask Bridge']).includes(
-        chainId,
-      ) ||
-      params.to.toLowerCase() !==
-        FIRST_PARTY_CONTRACT_NAMES['MetaMask Bridge'][chainId].toLowerCase()
-    ) {
+    const r = addrToExpMap[params.to.toLowerCase()];
+    // if undefined then no address matched
+    if (!r) {
+      return next();
+    }
+    const { experienceType, chainId: experienceChainId } = r;
+    // skip if chainId is different
+    if (experienceChainId !== chainId) {
```

Comment:
Some of the first party contracts use the same address on multiple chains, this means we can't assume an address correlates to a single chain ID.

Could we instead use a function such as `getExperience(address: Hex, chainId: Hex)` and verify we have a match?

We could also extend the `addrToExpMap` constant to include an array of chain IDs in the values, but that feels more complex and less readable.

Reply:
> Some of the first party contracts use the same address on multiple chains, this means we can't assume an address correlates to a single chain ID.

you're right. That's an oversight on my part. I think a getExperience will resolve this without reverse look up.